### PR TITLE
[IO-1692][internal] Made import more tolerant of potential errors

### PR DESCRIPTION
**Pull request by owencjones**
# Problem
In version 0.8.38, CX reported issues with intolerance of errors in some empty annotations, where these should simply be skipped.

# Solution
Added logic that checks for `None` values in the chained subscripted lookups, so that these will be logged and skipped in future.
Added use of logger for the output logging.

# Changelog
Added error checking block to importer

### CODE REVIEW

File: darwin/importer/importer.py [hunk 1]
```diff
@@ -39,6 +39,8 @@
 from darwin.utils import flatten_list, secure_continue_request
```
Comment (owencjones) [inline darwin/importer/importer.py:39]: **NB: Most changes are Black reformatting the file, I've marked the section involved**

File: darwin/importer/importer.py [hunk 2]
```diff
@@ -647,6 +645,16 @@ def _import_annotations(
     for annotation in annotations:
         annotation_class = annotation.annotation_class
         annotation_type = annotation_class.annotation_internal_type or annotation_class.annotation_type
+
```
Comment (owencjones) [inline darwin/importer/importer.py:648]: **Main body of changes here**

File: darwin/importer/importer.py [hunk 3]
```diff
@@ -26,7 +25,7 @@
 
 if TYPE_CHECKING:
     from darwin.client import Client
-    from darwin.dataset import RemoteDataset
+    from darwin.dataset.remote_dataset import RemoteDataset
```
Comment (Nathanjp91) [inline darwin/importer/importer.py:28]: Confused by this change? Was there a reason?

**Review by Nathanjp91 — APPROVED** [review]
Small q but not important

File: darwin/importer/importer.py (hunk 3 above)
Comment (owencjones) [reply darwin/importer/importer.py:28]: Mypy complaint - not direct import

**Pull request merged**